### PR TITLE
group and update some more dependencies, update gradle, update icon i…

### PR DESCRIPTION
**Pull request by TacoTheDank**
## General
- Update links in README.md
- Update icon to newer icon in README.md
- Add licenses in .travis.yml
- Set vectorDrawables.useSupportLibrary = true
- Shortened packagingOptions to just "exclude 'META-INF/*'"

## Dependency updates again
- Robolectric 3.3.1 -> 3.8
- PowerMockito 1.7.4 -> ~~2.0.0-RC.4~~ 2.0.0
- Play Services Drive 10.2.6 -> 11.2.0 (latest before breaking changes have to be made)
- Added Android Support ExifInterface
- "org.apache.commons:commons-lang3" 3.5 -> 3.8.1
- "org.apache.commons:commons-text" 1.4 -> 1.6
- "org.ligi:snackengage:0.5" updated to "com.github.ligi:snackengage:0.6"
- "com.github.rey5137:material" 1.2.4 -> 1.2.5
- "com.github.johnkil.android-robototextview:robototextview" changed to "com.github.johnkil:Android-RobotoTextView"
- "com.mikepenz:aboutlibraries" 5.8.5 -> 6.1.1
- "org.jetbrains:annotations-java5:15.0" has been changed to "org.jetbrains:annotations:16.0.3" (can be kept this way as long as app is compiled with Java 1.8, and there's no reason NOT to compile with Java 1.8, so...)
- "com.davemorrissey.labs:subsampling-scale-image-view" 3.6.0 -> 3.7.2
- "com.makeramen:roundedimageview" 2.2.1 -> 2.3.0
- "com.sothree.slidinguppanel:library" 3.3.1 -> 3.4.0
- "com.theartofdev.edmodo:android-image-cropper" 2.3.1 -> 2.7.0
- added note to replace "com.googlecode.mp4parser:isoparser" for newer library called "org.mp4parser:isoparser"
- Junit 4.12 -> 4.13-beta-1
- Mockito Core 2.23.0 -> 2.23.4

I've tested these new versions with compiled apks, and there aren't any issues I've come across :)

## All other files
- Basically just reformatted code, and converted the java expressions into lambdas in the test files

Signed-off-by: Taco <SkytkRSfan3895@gmail.com>

### CODE REVIEW

**Comment by TacoTheDank** [comment]
Moved "isFDroid" boolean to build.gradle's productFlavors; it's just easier configuration.

**Comment by TacoTheDank** [comment]
Actually, now that I look at this merge request and look through the commits, it's just a bit too messy for my taste.

**Comment by Tensounder54** [comment]
If you really want to do redo the PR then feel free but I've seen some pretty augur PR's and this is not one of themp
